File: Apps/EmployeeApp/src/app/model/vehicle.ts
export class Vehicle {
    constructor(
        private _id: number,
        private _status: string,
        private _image: string,
        private _manufacturer: string
    ) {}

    // Getter and Setter for id
    public get id(): number {
        return this._id;
    }

    public set id(id: number) {
        this._id = id;
    }

    // Getter and Setter for status
    public get status(): string {
        return this._status;
    }

    public set status(status: string) {
        this._status = status;
    }

    // Getter and Setter for image
    public get image(): string {
        return this._image;
    }

    public set image(image: string) {
        this._image = image;
    }

    // Getter and Setter for manufacturer
    public get manufacturer(): string {
        return this._manufacturer;
    }

    public set manufacturer(manufacturer: string) {
        this._manufacturer = manufacturer;
    }
}

File: RentVehicleServer/RentVehicleServer/src/main/java/server/vehicle/Vehicle.java
package server.vehicle;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Inheritance;
import jakarta.persistence.InheritanceType;
import jakarta.persistence.Lob;
import jakarta.persistence.Table;
import jakarta.persistence.Transient;

@Entity
@Table(name="vehicle")
@Inheritance(strategy = InheritanceType.JOINED)
public class Vehicle {

	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(name="id")
	private Integer id;
	
	@Column(name="status")
	private String status;
	@Lob
	@Column(name="image",columnDefinition = "MEDIUMBLOB")
	private byte[] image;
	
	@Column(name="deleted")
	private Boolean deleted=false;
	
	
	
	@Transient
	private String manufacturer;
	
	
	public Integer getId() {
		return id;
	}
	public void setId(Integer id) {
		this.id = id;
	}
	public String getStatus() {
		return status;
	}
	public void setStatus(String status) {
		this.status = status;
	}
	public byte[] getImage() {
		return image;
	}
	public void setImage(byte[] image) {
		this.image = image;
	}
	
	public String getManufacturer() {
		return manufacturer;
	}

	public void setManufacturer(String manufacturer) {
		this.manufacturer = manufacturer;
	}
	public Boolean getDeleted() {
		return deleted;
	}
	public void setDeleted(Boolean deleted) {
		this.deleted = deleted;
	}
	
	

}
